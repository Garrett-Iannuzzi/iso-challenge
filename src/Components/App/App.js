import React, { Component } from 'react';
import Nav from '../Nav/Nav';
import { Route } from 'react-router-dom';
import { getPlayers } from '../../apiCalls';
import './App.scss';

class App extends Component {
  constructor() {
    super()
    this.state = {
      players: []
    }
  }

  componentDidMount() {
    getPlayers(1)
      .then(res => this.setState({ players: res.data }))
  }

  render() {
    return (
      <main>
        <Route path='/' render={() => 
          <Nav /> 
          }
        />
      </main>
    );
  }
}

export default App;
