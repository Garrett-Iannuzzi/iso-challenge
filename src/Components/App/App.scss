main {
  background: url('../../images/sunset-bball.jpg');
  opacity: .9;
  background-size: cover;
  background-repeat: no-repeat;
  height: 100vh;
}