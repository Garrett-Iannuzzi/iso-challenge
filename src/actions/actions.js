export const getTeamsInfo = (teamOne, teamTwo) => ({
  type: 'GET_TEAMS',
  teamOne,
  teamTwo
});

export const getPlayerInfo = (playerInfo) => ({
  type: 'GET_PLAYERS',
  playerInfo
});

export const getStatsInfo = (stats) => ({
  type: 'GET_STATS',
  stats
});