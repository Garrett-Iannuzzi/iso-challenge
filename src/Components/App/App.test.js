import React from 'react';
import { App, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { getPlayerInfo } from '../../actions/actions';

describe('App', () => {

  let wrapper

  describe('App Component', () => {

    beforeEach(() => {
      wrapper = shallow(<App />)
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapDispatchToProps', () => {

    it('Should call mapDispatchToProps with correct information info when playerInfo is called', () => {
      const mockDispatch = jest.fn();
      const mockPlayerData = [
        {        
          id:14,
          first_name:"Ike",
          height_feet:null,
          height_inches:null,
          last_name:"Anigbogu",
          position:"C",
          team: {
            id:12,
            abbreviation:"IND",
            city:"Indiana",
            conference:"East",
            division:"Central",
            full_name:"Indiana Pacers",
            name:"Pacers"
          },
          weight_pounds:null
        },
        {
          id:25,
          first_name:"Ron",
          height_feet:null,
          height_inches:null,
          last_name:"Baker",
          position:"G",
          team: {
            id:20,
            abbreviation:"NYK",
            city:"New York",
            conference:"East",
            division:"Atlantic",
            full_name:"New York Knicks",
            name:"Knicks"
          },
          weight_pounds:null
        }
      ]
      const actionToDispatch = getPlayerInfo(mockPlayerData);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.playerInfo(mockPlayerData);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  });

})