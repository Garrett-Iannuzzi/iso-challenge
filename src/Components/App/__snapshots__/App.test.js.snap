// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component should match the snapshot 1`] = `
<body>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    path="/rules"
    render={[Function]}
  />
  <Route
    path="/game"
    render={[Function]}
  />
  <Route
    path="/score"
    render={[Function]}
  />
</body>
`;
